shopfloor_reception/services/reception.py

```diff
@@ -0,0 +1,1079 @@
+        return self._response_for_set_lot(picking, line)
+
+    def _select_line_from_packaging(self, picking, packaging):
+        line = fields.first(
+            picking.move_line_ids.filtered(
+                lambda l: l.package_id.product_packaging_id == packaging
+                and not l.result_package_id
+            )
+        )
+        if not line:
+            return self._response_for_select_line(
+                picking,
+                message=self.msg_store.packaging_not_found_in_picking(),
+            )
+        picking.user_id = self.env.user
+        if packaging.product_id.tracking not in ("lot", "serial"):
+            return self._response_for_set_quantity(picking, line)
+        return self._response_for_set_lot(picking, line)
+
+    def _order_stock_picking(self):
+        return "scheduled_date ASC"
+
+    def _scan_document__by_picking(self, barcode):
+        search = self._actions_for("search")
+        picking_filter_result = search.picking_from_scan(barcode, use_origin=True)
+        if picking_filter_result:
+            # There is a case where scanning the source document
+            # could return more than one picking.
+            # In this case, we ask the user to scan a package instead.
+            if len(picking_filter_result) > 1:
+                return self._response_for_select_document(
+                    pickings=picking_filter_result,
+                    message=self.msg_store.source_document_multiple_pickings_scan_package(),
+                )
+            return self._select_picking(picking_filter_result)
+
+    def _scan_document__by_product(self, barcode):
+        search = self._actions_for("search")
+        product = search.product_from_scan(barcode, use_packaging=False)
+        if product:
+            return self._select_document_from_product(product)
+
+    def _scan_document__by_packaging(self, barcode):
+        search = self._actions_for("search")
+        packaging = search.packaging_from_scan(barcode)
+        if packaging:
+            return self._select_document_from_packaging(packaging)
+
+    def _scan_line__by_product(self, picking, barcode):
+        search = self._actions_for("search")
+        product = search.product_from_scan(barcode, use_packaging=False)
+        if product:
+            return self._select_line_from_product(picking, product)
+
+    def _scan_line__by_packaging(self, picking, barcode):
+        search = self._actions_for("search")
+        packaging = search.packaging_from_scan(barcode)
+        if packaging:
+            return self._select_line_from_packaging(picking, packaging)
+
+    def _set_quantity__by_product(self, picking, selected_line, barcode):
+        # needed values:
+        # selected_line
+        # picking
+        search = self._actions_for("search")
+        product = search.product_from_scan(barcode, use_packaging=False)
+        if product:
+            selected_line.qty_done += 1
```

Comment:
We should check if the product scanned matches with the line, no? Right now I could scan product A while the one on the selected line is B.